### PR TITLE
fix issue with upgrading npm directly to 3.0.0 on node 0.8

### PR DESCRIPTION
**Pull request by calvinmetcalf**
fix for #158, built off of #156

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
but wait .travis is now suddenly failing for some unrelated reason

**Comment by calvinmetcalf** [comment]
part of this seems to be an npm issue npm/npm#9668 not sure about sauce labs

**Comment by calvinmetcalf** [comment]
and that require has a very good reason for being inside the constructor, it's because it avoids a circular dependency

**Comment by calvinmetcalf** [comment]
ping @nodejs/streams @dcousens

**Comment by mafintosh** [comment]
@calvinmetcalf where is that `require` line you're mentioning above? Otherwise LTGM

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 sudo: false
 language: node_js
 before_install:
+  - npm install -g npm@2
```
Comment (dcousens) [inline .travis.yml]: What are the implications of having both these lines?

File: test/common.js [hunk 2]
```diff
@@ -32,6 +32,7 @@ exports.libDir = path.join(exports.testDir, '../lib');
 exports.tmpDirName = 'tmp';
 exports.PORT = +process.env.NODE_COMMON_PORT || 12346;
 exports.isWindows = process.platform === 'win32';
+exports.isAix = process.platform === 'aix';
```
Comment (dcousens) [inline test/common.js]: What is this used by?

Comment (calvinmetcalf) [inline test/common.js]: no idea, this is from node.js and was added between 4.0.0 and 4.1.0

**Comment by calvinmetcalf** [comment]
@mafintosh that was here https://github.com/nodejs/readable-stream/blob/c2d1cce2a5000507c18481f703d28fe51ad7aab2/lib/_stream_writable.js#L59

File: .travis.yml (hunk 1 above)
Comment (calvinmetcalf) [inline .travis.yml]: so the version of npm that comes with node 0.8 breaks when it upgrades directly to npm@3, so it has to first upgrade to npm@2 and then from npm@2 upgrade to npm@3

**Comment by dcousens** [comment]
:+1:

**Pull request merged**